tests/foreman/ui/test_classparameters.py

```diff
@@ -0,0 +1,705 @@
+
+        @steps:
+        1.  Check an Override checkbox.
+        2.  Update the Key Type to yaml.
+        3.  Give a 'valid' default Value.
+        3.  Submit the changes.
+
+        @assert: Parameter Updated with yaml type successfully.
+
+        @status: Manual
+
+        """
+
+    @stubbed()
+    @run_only_on('sat')
+    @tier1
+    def test_negative_update_parameter_type_yaml(self):
+        """@test: Parameter not Updated for yaml type - Invalid value.
+
+        @feature: Class Parameters - Parameter Type
+
+        @steps:
+        1.  Check an Override checkbox.
+        2.  Update the Key Type to yaml.
+        3.  Give a 'Invalid' default Value.
+        3.  Submit the changes.
+
+        @assert: Parameter not updated with yaml type for invalid value.
+
+        @status: Manual
+
+        """
+
+    @stubbed()
+    @run_only_on('sat')
+    @tier1
+    def test_positive_update_parameter_type_json(self):
+        """@test: Parameter Update for json type.
+
+        @feature: Class Parameters - Parameter Type
+
+        @steps:
+        1.  Check an Override checkbox.
+        2.  Update the Key Type to json.
+        3.  Give a 'valid' default Value.
+        3.  Submit the changes.
+
+        @assert: Parameter Updated with json type successfully.
+
+        @status: Manual
+
+        """
+
+    @stubbed()
+    @run_only_on('sat')
+    @tier1
+    def test_negative_update_parameter_type_json(self):
+        """@test: Parameter not Updated for json type - Invalid value.
+
+        @feature: Class Parameters - Parameter Type
+
+        @steps:
+        1.  Check an Override checkbox.
+        2.  Update the Key Type to json.
+        3.  Give a 'Invalid' default Value.
+        3.  Submit the changes.
+
+        @assert: Parameter not updated with json type for invalid value.
+
+        @status: Manual
+
+        """
+
+    @stubbed()
+    @run_only_on('sat')
+    @tier3
+    def test_positive_validate_puppet_default_value(self):
+        """@test: Validation doesn't works on puppet default value.
+
+        @feature: Class Parameters - Optional Validation
+
+        @steps:
+        1.  Check an Override checkbox.
+        2.  Check 'Use Puppet Default' value.
+        3.  Validate this value under section 'Optional Input Validator'.
+
+        @assert: Validation doesn't works with puppet default value.
+
+        @status: Manual
+
+        """
+
+    @stubbed()
+    @run_only_on('sat')
+    @tier3
+    def test_positive_validate_default_value_required_checkbox(self):
+        """@test: Error raised for blank default Value - Required checkbox.
+
+        @feature: Class Parameters - Optional Validation
+
+        @steps:
+        1.  Check an Override checkbox.
+        2.  Don't provide any default value, keep blank.
+        3.  Check Required checkbox in 'Optional Input Validator'.
+        4.  Submit the change.
+
+        @assert: Error raised for blank default value by 'Required' checkbox.
```

Comment:
This is been addressed by changing the Test Case name to negative and below one to negative.